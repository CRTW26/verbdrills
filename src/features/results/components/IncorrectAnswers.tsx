import React from 'react'
import { IncorrectAnswer } from 'shared/types'

interface Props {
  incorrectAnswers: IncorrectAnswer[]
}

const IncorrectAnswers: React.FC<Props> = ({ incorrectAnswers }) => {
  return (
    <>
      <h3>Incorrect answers</h3>
      <div className="row">
        <p>Infinitive</p>
        <p>Person</p>
        <p>Your answer</p>
        <p>Correct answers</p>
      </div>

      <>
        {Array.isArray(incorrectAnswers) &&
          incorrectAnswers.map((answer, index) => (
            <div key={index} className="row">
              <p>{answer.infinitive}</p>
              <p>{answer.person}</p>
              <p>{answer.input}</p>
              <p>{answer.answer}</p>
            </div>
          ))}
      </>

      <style jsx>{`
        .row {
          display: flex;
        }

        .row > p {
          flex: 0 1 25%;
          padding: 0.5rem 1rem 0.5rem 0;
        }
      `}</style>
    </>
  )
}

export default IncorrectAnswers
