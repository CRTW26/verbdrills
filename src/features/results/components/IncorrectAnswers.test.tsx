import { render } from '@testing-library/react'
import IncorrectAnswers from './IncorrectAnswers'

describe('<IncorrectAnswers />', () => {
  let props

  beforeEach(() => {
    props = {
      incorrectAnswers: [
        {
          infinitive: 'hablar',
          person: 'yo',
          input: 'hable',
          answer: 'hablo',
        },
      ],
    }
  })

  test('should render', () => {
    const { asFragment } = render(<IncorrectAnswers {...props} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
