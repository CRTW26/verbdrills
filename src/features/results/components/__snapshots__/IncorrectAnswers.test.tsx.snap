// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IncorrectAnswers /> should render 1`] = `
<DocumentFragment>
  <h3
    class="jsx-762435235"
  >
    Incorrect answers
  </h3>
  <div
    class="jsx-762435235 row"
  >
    <p
      class="jsx-762435235"
    >
      Infinitive
    </p>
    <p
      class="jsx-762435235"
    >
      Person
    </p>
    <p
      class="jsx-762435235"
    >
      Your answer
    </p>
    <p
      class="jsx-762435235"
    >
      Correct answers
    </p>
  </div>
  <div
    class="jsx-762435235 row"
  >
    <p
      class="jsx-762435235"
    >
      hablar
    </p>
    <p
      class="jsx-762435235"
    >
      yo
    </p>
    <p
      class="jsx-762435235"
    >
      hable
    </p>
    <p
      class="jsx-762435235"
    >
      hablo
    </p>
  </div>
</DocumentFragment>
`;
